commands/bake.go

```diff
@@ -108,7 +109,28 @@ func runBake(dockerCli command.Cli, targets []string, in bakeOptions) (err error
 		return err
 	}
 
-	_, err = build.Build(ctx, dis, bo, dockerAPI(dockerCli), dockerCli.ConfigFile(), printer)
+	resp, err := build.Build(ctx, dis, bo, dockerAPI(dockerCli), dockerCli.ConfigFile(), printer)
+	if err != nil {
+		return err
+	}
+
+	if len(in.metadataFile) > 0 && resp != nil {
+		if err := os.Remove(in.metadataFile); err != nil && !os.IsNotExist(err) {
```

Comment:
Could we do this with `os.Rename` instead or with `AtomicWriteFile()`. Is this for symlink protection? Otherwise, I don't think you need it at all.

Reply:
> with `AtomicWriteFile()`

SGTM